sections/testingandquality/refactoring.md

```diff
@@ -0,0 +1,27 @@
+# Refactoring
+
+<br/><br/>
+
+### Motivation for Refactoring
+
+Refactoring is an important process in the iterative development flow. Removing "Code Smells" (bad coding practices) such as Duplicated Code, Long Methods, Long Parameter list will improve your code and making it more maintainable. Using a static analysis tools will assist you in finding these code smells and build a process around refactoring.
+<br/><br/>
+
+
```

Comment:
@TheHollidayInn I suggest adding an image or two of a cool analysis report (e.g. SonarQube dashboard)

Reply:
Will do!